service/storage_proxy.cc

```diff
@@ -3496,12 +3496,13 @@ class abstract_read_executor : public enable_shared_from_this<abstract_read_exec
 protected:
     future<> make_mutation_data_requests(lw_shared_ptr<query::read_command> cmd, data_resolver_ptr resolver, targets_iterator begin, targets_iterator end, clock_type::time_point timeout) {
         return parallel_for_each(begin, end, [this, &cmd, resolver = std::move(resolver), timeout] (gms::inet_address ep) {
-            return make_mutation_data_request(cmd, ep, timeout).then_wrapped([this, resolver, ep] (future<rpc::tuple<foreign_ptr<lw_shared_ptr<reconcilable_result>>, cache_temperature>> f) {
+            return make_mutation_data_request(cmd, ep, timeout).then_wrapped([this, resolver, ep, start = clock_type::now()] (future<rpc::tuple<foreign_ptr<lw_shared_ptr<reconcilable_result>>, cache_temperature>> f) {
```

Comment:
Nit: You can do `start = clock_type::now()` only once, before calling `parallel_for_each`, and pass the `start` value down the chain.

As I can see, here `clock_type == lowres_clock`, so it's very likely that all invocations of `clock_type::now()` here will return the same value because of the 10ms granularity. On the other hand, `clock_type::now()` involves an atomic operation, so reducing the number of calls to it is a small win.

The same can be done for `make_data_requests` and `make_digest_request`.

Reply:
It's a cheap atomic operation (with relaxed memory order), equivalent to any other memory read. Atomicity here only prevents the read from being split into two subword reads, which the compiler wouldn't do anyway.